react-native/react/constants/platform.native.desktop.js

```diff
@@ -20,26 +20,43 @@ const envedPathOSX = {
   prod: 'Keybase'
 }
 
+function buildWin32SocketRoot() {
+  let appdata = process.env.APPDATA
+  // Remove leading drive letter e.g. C:
+  if (/^[a-zA-Z]:/.test(appdata)) {
+    appdata = appdata.slice(2)
+  }
+  // Handle runModes, prod has no extension.
+  let extension = ''
+  if (runMode !== 'production') {
+    extension = runMode.charAt(0).toUpperCase() + runMode.substr(1)
+  }
+  let path = `\\\\.\\pipe\\kbservice${appdata}\\Keybase${extension}`
+  return path
+}
```

Comment:
The fix here is actually to *remove* the service's dynamism.  The GUI does not share environment variables with the service, or know where on the filesystem its config file is to parse it -- there needs to be a contract that the service will follow a specified algorithm for placing the socket file and the GUI will follow the same one.

(So the GUI similarly ignores env/config on OS X, it's not new to Windows.)